exercises/custom-set/custom_set_test.rb

```diff
@@ -3,114 +3,303 @@
+  end
+
+  def test_when_the_element_is_in_the_set
+    skip
+    set = CustomSet.new [1, 2, 3]
+    element = 1
+    assert set.member? element
+  end
+
+  def test_when_the_element_is_not_in_the_set
+    skip
+    set = CustomSet.new [1, 2, 3]
+    element = 4
+    refute set.member? element
+  end
+
+  def test_empty_set_is_a_subset_of_another_empty_set
+    skip
+    set1 = CustomSet.new []
+    set2 = CustomSet.new []
+    assert set2.subset? set1
+  end
+
+  def test_empty_set_is_a_subset_of_non_empty_set
+    skip
+    set1 = CustomSet.new []
+    set2 = CustomSet.new [1]
+    assert set2.subset? set1
+  end
+
+  def test_non_empty_set_is_not_a_subset_of_empty_set
+    skip
+    set1 = CustomSet.new [1]
+    set2 = CustomSet.new []
+    refute set2.subset? set1
+  end
+
+  def test_set_is_a_subset_of_set_with_exact_same_elements
+    skip
+    set1 = CustomSet.new [1, 2, 3]
+    set2 = CustomSet.new [1, 2, 3]
+    assert set2.subset? set1
+  end
+
+  def test_set_is_a_subset_of_larger_set_with_same_elements
+    skip
+    set1 = CustomSet.new [1, 2, 3]
+    set2 = CustomSet.new [4, 1, 2, 3]
+    assert set2.subset? set1
+  end
+
+  def test_set_is_not_a_subset_of_set_that_does_not_contain_its_elements
+    skip
+    set1 = CustomSet.new [1, 2, 3]
+    set2 = CustomSet.new [4, 1, 3]
+    refute set2.subset? set1
+  end
+
+  def test_the_empty_set_is_disjoint_with_itself
+    skip
+    set1 = CustomSet.new []
+    set2 = CustomSet.new []
+    assert set1.disjoint? set2
+  end
+
+  def test_empty_set_is_disjoint_with_non_empty_set
+    skip
+    set1 = CustomSet.new []
+    set2 = CustomSet.new [1]
+    assert set1.disjoint? set2
   end
 
-  def test_no_duplicates
-    assert_equal CustomSet.new([1, 1]), CustomSet.new([1])
+  def test_non_empty_set_is_disjoint_with_empty_set
+    skip
+    set1 = CustomSet.new [1]
+    set2 = CustomSet.new []
+    assert set1.disjoint? set2
   end
 
-  def test_delete
+  def test_sets_are_not_disjoint_if_they_share_an_element
     skip
-    assert_equal CustomSet.new([1, 3]), CustomSet.new([3, 2, 1]).delete(2)
-    assert_equal CustomSet.new([1, 2, 3]), CustomSet.new([3, 2, 1]).delete(4)
-    assert_equal CustomSet.new([1, 2, 3]), CustomSet.new([3, 2, 1]).delete(2.0)
-    assert_equal CustomSet.new([1, 3]), CustomSet.new([3, 2.0, 1]).delete(2.0)
+    set1 = CustomSet.new [1, 2]
+    set2 = CustomSet.new [2, 3]
+    refute set1.disjoint? set2
   end
 
-  def test_difference
+  def test_sets_are_disjoint_if_they_share_no_elements
     skip
-    assert_equal CustomSet.new([1, 3]),
-                 CustomSet.new([1, 2, 3]).difference(CustomSet.new([2, 4]))
+    set1 = CustomSet.new [1, 2]
+    set2 = CustomSet.new [3, 4]
+    assert set1.disjoint? set2
+  end
 
-    assert_equal CustomSet.new([1, 2.0, 3]),
-                 CustomSet.new([1, 2.0, 3]).difference(CustomSet.new([2, 4]))
+  def test_empty_sets_are_equal
+    skip
+    set1 = CustomSet.new []
+    set2 = CustomSet.new []
+    assert_equal set1, set2
```

Comment:
This is the right place to test equality. Note that the first test (about emptiness) is actually more complex than this as its a duplicate except it requires no args to `new` to be treated the same as an empty array to new.